### PR TITLE
Fix ds18 temperature conversion

### LINKED ISSUE #13287: Driver for DS18B20 returns invalid values
#### Description

`ds18_get_temperature` from `ds18` driver returns invalid values.

#### Steps to reproduce the issue

Hardware: Arduino Nano (ATMega328p) and DS18B20 temperature sensor. Code to reproduce the problem:

```
#include <stdio.h>
#include "board.h"
#include "ds18.h"
#include "xtimer.h"

#define SAMPLING_PERIOD 5

static const ds18_params_t ds18_params = {
    .pin = ARDUINO_PIN_2,
    .out_mode = GPIO_OD,
    .in_mode = GPIO_IN,
};

static ds18_t dev;
int16_t temperature;

int main()
{
    printf("Init DS18B20...");

    if (ds18_init(&dev, &ds18_params) == DS18_ERROR)
    {
        puts("FAILED");
        return 1;
    }
    else
    {
        puts("OK");
    };

    while (1)
    {
        if (ds18_get_temperature(&dev, &temperature) == DS18_OK)
        {
            printf("%d\n", temperature);
        }
        else
        {
            puts("Could not read temperature");
        };

        xtimer_sleep(SAMPLING_PERIOD);
    }

    return 0;
}
```

#### Expected results

In my room 25°C, but `ds18_get_temperature` returns `-77` (should return 2500).

### PR DESCRIPTION
**Pull request by catcombo**
### Contribution description

Fix `ds18` temperature conversion for issue #13287

### Testing procedure

I tested PR on my hardware (Arduino Nano ATmega328p and DS18B20 sensor). Also I tests for different input values to cover boundary values, negative and positive values and [related issues](https://github.com/RIOT-OS/RIOT/commit/1783b3da55902431198a295560cc4a473aa711ad).

### Issues/PRs references

Fixes #13287

### CODE REVIEW

File: drivers/ds18/ds18.c [hunk 1]
```diff
@@ -184,8 +184,8 @@ int ds18_read(const ds18_t *dev, int16_t *temperature)
 
     DEBUG("[DS18] Received byte: 0x%02x\n", b2);
 
-    int32_t measurement = ((int16_t)(b2 << 8 | b1) * 625);
-    *temperature = (int16_t)(measurement / 100);
+    int16_t measurement = (int16_t)(b2 << 8 | b1);
```
Comment (benpicco) [inline drivers/ds18/ds18.c]: Since you are casting to `int32_t` anyway (and that being it's only use) you might as well use `int32_t` here and drop the cast - it should not make a difference for the generated code.

**Comment by catcombo** [comment]
@benpicco Thank you for review! I don't write on C at my work, so I may be inaccurate in my code. Also I will try to be more attentive in the future and leave correct comments to my commits.

**Comment by catcombo** [comment]
Should I squash the commits?

**Comment by benpicco** [comment]
Yes please do.

**Comment by catcombo** [comment]
@leandrolanzieri Thank you. Done.
